src/EuroDNS/Helper/EuroDNSApi.php

```diff
@@ -0,0 +1,1793 @@
+     */
+    public function startsWith($haystack, $needle)
+    {
+        // Check if the PHP version is 8 or above to use the built-in function
+        if (phpversion() >= 8) {
+            // Use str_starts_with function if PHP version is 8 or above
+            return str_starts_with($haystack, $needle);
+        }
+
+        // Use the traditional substr method for older PHP versions
+        return substr($haystack, 0, strlen($needle)) === $needle;
+    }
+
+    /**
+     * Get the formatted phone number with the correct extension based on the country code.
+     *
+     * @param string $code              The country code.
+     * @param array  $countryCodeArray  The array containing the country code information.
+     * @param string $phoneNumber       The original phone number.
+     *
+     * @return string The formatted phone number with the correct extension.
+     */
+    private function getPhoneExtension($code, $countryCodeArray, $phoneNumber)
+    {
+        // Initialize the result with the original phone number
+        $res = $phoneNumber;
+
+        // Check if the phone number starts with '00' followed by the country code
+        if ($this->startsWith($res, '00' . $countryCodeArray['code'])) {
+            // Calculate the length of '00' followed by the country code
+            $length = strlen('00' . $countryCodeArray['code']);
+
+            // Return the formatted phone number with the correct extension
+            return '+' . $countryCodeArray['code'] . substr($res, $length);
+        }
+
+        // Special case for Germany (DE)
+        if ('DE' === $code) {
+            // Check if the phone number starts with '0'
+            if ($this->startsWith($res, '0')) {
+                // Return the formatted phone number with the correct extension
+                return '+' . $countryCodeArray['code'] . substr($res, 1);
+            }
+
+            // Return the formatted phone number with the correct extension
+            return '+' . $countryCodeArray['code'] . $res;
+        }
+
+        // For other countries, return the formatted phone number with the correct extension
+        return '+' . $countryCodeArray['code'] . $res;
+    }
+
+
+
+    /**
+     * FUNCTION connect
+     * Connects to the API and returns XML response.
+     *
+     * @param string $request The XML request to be sent to the API.
+     *
+     * @return mixed $response The XML response from the API or false if an error occurs.
+     */
+    private function connect(string $request)
+    {
+        // Check if URL, USERNAME, and PASSWORD constants are defined
+        if (!defined('URL')) {
+
+            // Load configuration from a file
+            $configFile = __DIR__ . '/../Data/config.json';
+            $config = json_decode(file_get_contents($configFile));
+
+            // Check the sandbox option is set or not
+            if ($this->configuration->sandbox) {
+                define('URL', $config->url_sandbox);
```

Comment:
Please do not define global constants. Depending on the runtime, the provision system may execute many provision requests in a single process, so using global constants is not an option.

I'd suggest rather than using a config.json file, to just set the 2 urls as class constants, and then set the username/password as private properties on this object instance.

Reply:
Please, we need to remove all global constants and just use normal variables or class properties instead:
- URL
- USERNAME
- PASSWORD

This is because processes may be re-used between provision requests so the use of global constants can cause real problems.

Reply:
updated